src/LiteCore/test/LiteCore.Tests.Shared/ScopeCollectionTest.cs

```diff
@@ -0,0 +1,574 @@
+                    arrColl = Native.c4db_collectionNames(Db, scopeName + 1);
+                    for (uint i = 1; i <= Native.FLArray_Count((FLArray*)arrColl); i++)
+                    {
+                        var collStr = (string)FLSliceExtensions.ToObject(Native.FLArray_Get((FLArray*)arrColl, i - 1));
+                        collStr.Should().Be(collName + (i + 1), "Because the deleted collection is no longer found in scope.");
+                        cnt++;
+                    }
+
+                    cnt.Should().Be(1, "Because collection collName1 is deleted.");
+                    Native.FLValue_Release((FLValue*)arrColl);
+                };
+            });
+        }
+
+        [Fact]
+        public void TestDefaultCollectionInfo()
+        {
+            RunTestVariants(() => {
+                Native.c4coll_getDocumentCount(DefaultColl).Should().Be(0, "because the default collection is empty");
+                Native.c4coll_getLastSequence(DefaultColl).Should().Be(0, "because the default collection is empty");
+                var publicID = new C4UUID();
+                var privateID = new C4UUID();
+                C4Error err;
+                var uuidSuccess = Native.c4db_getUUIDs(Db, &publicID, &privateID, &err);
+                if (!uuidSuccess) {
+                    throw CouchbaseException.Create(err);
+                }
+
+                var p1 = publicID;
+                var p2 = privateID;
+                var match = true;
+                for (int i = 0; i < C4UUID.Size; i++) {
+                    if (publicID.bytes[i] != privateID.bytes[i]) {
+                        match = false;
+                        break;
+                    }
+                }
+
+                match.Should().BeFalse("because public UUID and private UUID should differ");
+                (p1.bytes[6] & 0xF0).Should().Be(0x40, "because otherwise the UUID is non-conformant");
+                (p1.bytes[8] & 0xC0).Should().Be(0x80, "because otherwise the UUID is non-conformant");
+                (p2.bytes[6] & 0xF0).Should().Be(0x40, "because otherwise the UUID is non-conformant");
+                (p2.bytes[8] & 0xC0).Should().Be(0x80, "because otherwise the UUID is non-conformant");
+
+                // Make sure the UUIDs are persistent
+                ReopenDB();
+                var publicID2 = new C4UUID();
+                var privateID2 = new C4UUID();
+                uuidSuccess = Native.c4db_getUUIDs(Db, &publicID2, &privateID2, &err);
+                if (!uuidSuccess) {
+                    throw CouchbaseException.Create(err);
+                }
+
+                for (int i = 0; i < C4UUID.Size; i++) {
+                    publicID2.bytes[i].Should().Be(publicID.bytes[i]);
+                    privateID2.bytes[i].Should().Be(privateID.bytes[i]);
+                }
+            });
+        }
+
+        #endregion
+
+        #region Document
+
+        [Fact]
+        public void TestCreateVersionedDoc()
```

Comment:
Not sure this test is worth having.  Revisit when the implementation is done to see if c4coll_putDoc is used anywhere.  I think it is going to be limited to c4coll_createDoc and c4doc_update, as in the test following this one.